components/formats-bsd/src/loci/formats/in/FilePatternReader.java

```diff
@@ -331,12 +336,42 @@ public void close() throws IOException {
   @Override
   protected void initFile(String id) throws FormatException, IOException {
     super.initFile(id);
-    // read the pattern from the file
-    // the file should just contain a single line with the relative or
-    // absolute file pattern
-
     currentId = new Location(id).getAbsolutePath();
-    pattern = DataTools.readFile(id).trim();
+
+    Map<String, String> pairs = new HashMap<String, String>();
+    BufferedReader br = new BufferedReader(new FileReader(id));
+    String line;
+    while ((line = br.readLine()) != null) {
+      line = line.trim();
+      if (line.length() == 0) {
+        continue;
+      }
+      if (line.startsWith("#")) {
```

Comment:
I got into the habit of using `startsWith` because it also works when the string is empty. Here, however, we've already check for non-emptiness, so I can make the change.